File: benchmarkDS/ZeroShot/OPENAI_gpt-5-2025-08-07/Lse7.py
import sys
from typing import List, Optional, Tuple

def try_parse_non_negative_int(s: str) -> Tuple[bool, int]:
    if s is None:
        return False, 0
    t = s.strip()
    if not t:
        return False, 0
    if t[0] == '+':
        t = t[1:]
        if not t:
            return False, 0
    if t[0] == '-':
        return False, 0
    if not all('0' <= ch <= '9' for ch in t):
        return False, 0
    val = 0
    for ch in t:
        val = val * 10 + (ord(ch) - 48)
        if val > 2**31 - 1:
            return False, 0
    return True, val

def get_arg_at_index(args: List[str], index_str: str) -> Optional[str]:
    ok, idx = try_parse_non_negative_int(index_str)
    if not ok:
        return None
    if idx < 0 or idx >= len(args):
        return None
    return args[idx]

def print_arg_at_index(args: List[str], index_str: str) -> None:
    v = get_arg_at_index(args, index_str)
    if v is not None:
        print(v)

def main():
    # If run with CLI arguments, use the second argument as index, and print the argument at that index.
    if len(sys.argv) > 1:
        print_arg_at_index(sys.argv, sys.argv[1])

    # 5 test cases
    tests = [
        ["a0", "2", "x0", "x1", "x2"],        # prints x0
        ["first", "0", "z"],                  # prints first
        ["foo", "3", "aa", "bb", "cc", "dd"], # prints dd
        ["unused", "-1", "val"],              # prints nothing (invalid)
        ["hello", "abc", "p", "q"],           # prints nothing (invalid)
    ]
    for t in tests:
        print_arg_at_index(t, t[1])

if __name__ == "__main__":
    main()

File: benchmarkDS/ourMethod/OPENAI_gpt-5-2025-08-07/Lse7.java
import java.nio.charset.StandardCharsets;
import java.util.Arrays;

public class Lse7 {
    // Converts a non-negative decimal string to int with overflow and format checks.
    // Returns null if invalid.
    public static Integer stringToInt(String s) {
        if (s == null) return null;
        String t = s.trim();
        if (t.isEmpty()) return null;
        int i = 0;
        int n = t.length();

        // Optional leading '+'
        char c0 = t.charAt(0);
        if (c0 == '+') {
            i++;
            if (i >= n) return null; // "+" only is invalid
        } else if (c0 == '-') {
            return null; // negative indices not allowed
        }

        int result = 0;
        while (i < n) {
            char ch = t.charAt(i++);
            if (ch < '0' || ch > '9') return null;
            int digit = ch - '0';
            if (result > (Integer.MAX_VALUE - digit) / 10) {
                return null; // overflow
            }
            result = result * 10 + digit;
        }
        return result;
    }

    // Prints the argument at the INDEX specified by indexStr within args.
    // Returns true on success, false on validation failure.
    public static boolean printArgAtIndex(String[] args, String indexStr) {
        if (args == null || args.length == 0 || indexStr == null) {
            System.err.println("ERROR");
            return false;
        }
        Integer idx = stringToInt(indexStr);
        if (idx == null) {
            System.err.println("ERROR");
            return false;
        }
        if (idx < 0 || idx >= args.length) {
            System.err.println("ERROR");
            return false;
        }
        // Print precisely the target argument
        System.out.println(args[idx]);
        return true;
    }

    private static void runTests() {
        // 5 test cases
        String[][] tests = new String[][]{
            new String[]{"x", "2", "a", "b", "c", "d"},       // prints "a"
            new String[]{"index", "0", "hello"},              // prints "index"
            new String[]{"index", "3", "p", "q", "r", "s"},   // prints "s"
            new String[]{"cmd", "100", "a", "b"},             // out of bounds -> ERROR (stderr)
            new String[]{"cmd", "abc", "z"}                   // invalid index -> ERROR (stderr)
        };
        for (String[] t : tests) {
            // the second argument (index 1) is the index string
            if (t.length > 1) {
                printArgAtIndex(t, t[1]);
            } else {
                System.err.println("ERROR");
            }
        }
    }

    public static void main(String[] args) {
        // Run 5 test cases
        runTests();

        // If invoked with real command line arguments, also process them:
        // Here, 'args[1]' (second argument) is the index string.
        if (args != null && args.length > 1) {
            printArgAtIndex(args, args[1]);
        }
    }
}